File: 4th Semester/Mid/Factorial.java
package Mid;/* Factiorial Number */

public class Factorial {

    public static void main(String[] args) {
        int n = 6, fact = 1;

        for(int i=1; i<=n; i++){
            //Mid.Factorial formula
            fact = fact*i;
        }

        System.out.println("Mid.Factorial of 6 is: " + fact);
    }

}

File: Recursion/factorial.c
// Base Case where this function will stop
// Recursive Call to do sam things untill base case

// Lets Start Codding

#include <stdio.h>

// create factorial function
int fact(int num)
{
    int sum;
    // base case
    if (num == 1)
    {
        return 1;
    }
    else
    {
        sum = num * fact(num - 1);
    }
    return sum;
}

int main()
{
    int num, result;
    printf("Please enter a number: ");
    scanf("%d", &num);
    result = fact(num);
    printf("Factorial of %d is: %d", num, result);

    return 0;
}